### PR TITLE
Introduce agent-ci-experience and agent-developer-tools

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Introduce agent-developer-tools and agent-ci-experience, replacing agent-platform

### Motivation

agent-platform team became agent-developer-tools, agent-ci-experience and agent-build-and-releases. This PR moves ownership across `datadog-agent`

### CODE REVIEW

File: .gitlab/kitchen_testing/new-e2e_testing/ubuntu.yml [hunk 1]
```diff
@@ -6,13 +6,13 @@
 .new-e2e_install_script:
   variables:
     TARGETS: ./tests/agent-platform/install-script
-    TEAM: agent-platform
+    TEAM: agent-build-and-releasesagent-platform
```
Comment (spencergilbert) [inline .gitlab/kitchen_testing/new-e2e_testing/ubuntu.yml:9]:
```suggestion
    TEAM: agent-build-and-releases
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]

**Review by spencergilbert — APPROVED** [review]

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -149,7 +149,7 @@
 /cmd/agent/dist/conf.d/containerd.d/    @DataDog/container-integrations
 /cmd/agent/dist/conf.d/container_image.d/      @DataDog/container-integrations
 /cmd/agent/dist/conf.d/container_lifecycle.d/  @DataDog/container-integrations
-/cmd/agent/dist/conf.d/jetson.d/        @DataDog/agent-platform
+/cmd/agent/dist/conf.d/jetson.d/        @DataDog/@DataDog/agent-ci-experience
```
Comment (KSerrania) [inline .github/CODEOWNERS:152]:
```suggestion
/cmd/agent/dist/conf.d/jetson.d/        @DataDog/agent-ci-experience
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -51,30 +51,30 @@
...
 /.github/workflows/buildimages-update.yml           @DataDog/agent-build-and-releases @DataDog/agent-shared-components
 
-/.run                                               @DataDog/agent-platform
+/.run                                               @DataDog/agent-developer-tools
 /.run/docker/                                       @DataDog/container-integrations
 
 # Gitlab files
-# Files containing job contents are owned by teams in charge of the jobs + agent-platform
-# Files that only describe structure (eg. includes, rules) are owned by agent-platform
+# Files containing job contents are owned by teams in charge of the jobs + agent-ci-experience
```
Comment (KSerrania) [inline .github/CODEOWNERS:58]:
```suggestion
# Files containing job contents are owned by teams in charge of the jobs + agent-ci-experience or agent-build-and-releases
```

File: .github/CODEOWNERS (hunk 2 above)
Comment (KSerrania) [inline .github/CODEOWNERS:152]: Note: to be updated later, it doesn't make sense for this to be owned by any of the ex-Agent Platform teams.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -329,14 +329,14 @@
 /pkg/collector/corechecks/system/winkmem/         @DataDog/windows-agent
 /pkg/collector/corechecks/system/winproc/         @DataDog/windows-agent
 /pkg/collector/corechecks/systemd/      @DataDog/agent-integrations
-/pkg/collector/corechecks/nvidia/       @DataDog/agent-platform
+/pkg/collector/corechecks/nvidia/       @DataDog/agent-ci-experience
```
Comment (KSerrania) [inline .github/CODEOWNERS:332]: Note: to be updated later, it doesn't make sense for this to be owned by any of the ex-Agent Platform teams.

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -329,14 +329,14 @@
...
 /pkg/collector/corechecks/windows_event_log/ @DataDog/windows-agent
 /pkg/config/                            @DataDog/agent-shared-components
 /pkg/config/config_template.yaml        @DataDog/agent-shared-components @DataDog/documentation
 /pkg/config/setup/apm.go                @DataDog/agent-apm
 /pkg/config/autodiscovery/              @Datadog/container-integrations
 /pkg/config/env                         @DataDog/container-integrations @DataDog/container-app
-/pkg/config/logs                        @Datadog/agent-shared-components @Datadog/agent-platform
+/pkg/config/logs                        @Datadog/agent-shared-components @Datadog/agent-ci-experience
```
Comment (KSerrania) [inline .github/CODEOWNERS:339]: Note: I don't think it makes sense for Agent Platform to own this directory at all.

**Review by KSerrania — APPROVED** [review]

File: .github/CODEOWNERS (hunk 5 above)
Comment (pducolin) [reply .github/CODEOWNERS:339]:
```suggestion
/pkg/config/logs                        @Datadog/agent-shared-components
```

**Review by KevinFairise2 — APPROVED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (pducolin) [reply .github/CODEOWNERS]: Posted in `datadog-agent` slack to find someone to adopt it

**Pull request merged**